src/rpc/evo.cpp

```diff
@@ -746,41 +741,59 @@ static UniValue protx_register_common_wrapper(const JSONRPCRequest& request,
     } else {
         // referencing external collateral
 
-        Coin coin;
-        if (!GetUTXOCoin(ptx.collateralOutpoint, coin)) {
-            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("collateral not found: %s", ptx.collateralOutpoint.ToStringShort()));
-        }
-        CTxDestination txDest;
-        ExtractDestination(coin.out.scriptPubKey, txDest);
-        const PKHash *pkhash = std::get_if<PKHash>(&txDest);
-        if (!pkhash) {
-            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, strprintf("collateral type not supported: %s", ptx.collateralOutpoint.ToStringShort()));
+        bool unlockOnError = false;
+        {
+            LOCK(wallet->cs_wallet);
+            if (!wallet->IsLockedCoin(ptx.collateralOutpoint.hash, ptx.collateralOutpoint.n)) {
+                wallet->LockCoin(ptx.collateralOutpoint);
+                unlockOnError = true;
+            }
         }
```

Comment:
How about this to make it const
```suggestion
        const bool unlockOnError = [&]() {
            if (LOCK(wallet->cs_wallet); !wallet->IsLockedCoin(ptx.collateralOutpoint.hash, ptx.collateralOutpoint.n)) {
                wallet->LockCoin(ptx.collateralOutpoint);
                return true;
            }
        }();
```